### ISSUE TITLE
Accuracy keeps increasing with more epochs

### ISSUE THREAD

**Issue by cmavro** [VITA-Group/Large_Scale_GCN_Benchmarking#3]
Hi, very interesting work!

I am experimenting with EnGCN, but realized that the accuracy results (train / val / test) keep increasing when we have more epochs. For example for the script attached for ogbn-arxiv
[EnGCN_v2.txt](https://github.com/VITA-Group/Large_Scale_GCN_Benchmarking/files/10079143/EnGCN_v2.txt), 
I am getting the following results per stage:
```
Model: 00, Epoch: 499, Train acc: 77.6185, Valid acc: 58.3073, Test acc: 56.0274
------ pseudo labels updated, rate: 0.8135 ------
Model: 01, Epoch: 499, Train acc: 84.5318, Valid acc: 75.2710, Test acc: 74.1765
------ pseudo labels updated, rate: 0.9548 ------
Model: 02, Epoch: 499, Train acc: 91.2519, Valid acc: 82.7779, Test acc: 82.2418
------ pseudo labels updated, rate: 0.9851 ------
Model: 03, Epoch: 499, Train acc: 86.0443, Valid acc: 82.8652, Test acc: 81.8591
------ pseudo labels updated, rate: 0.9903 ------
Final train acc: 88.1417, Final valid acc: 81.9960, Dianl test acc: 80.8016
```
Is this normal to happen? Actually, Model 00 has a steady accuracy after ~100 epochs, but Model 01-03 accuracy keeps increasing with more epochs. Testing this on ogbn-products yields similar results. 
Did you also observe this during hyper-parameter search? What happens when you try more epochs?

I am also attaching my environment's specifications 
[myreqs2.txt](https://github.com/VITA-Group/Large_Scale_GCN_Benchmarking/files/10079158/myreqs2.txt).

Looking forward to your response!

**Comment by cmavro** [comment]
btw, this does not happen when I deactivate SLE, i.e., SLE_threshold >=1.

**Comment by vermouthdky** [comment]
@cmavro yes, It did happen when you tried to run more epochs. Probably it is due to the self-learning part we used as it may generate pesudo labels that are consistent with the ground truth with high probability and the corresponding MLP will try its best to fit it.

Model 00 has a steady accuracy because it does not utilize any label information, as introduced in our paper.

**Comment by vermouthdky** [comment]
> 

BTW, we set a fixed number of epoch in order to have a fair comparison with other baselines. You may have a better performance if you tune the hyperparameters with a wider searching space.
